fvm/environment/event_emitter.go

```
@@ -201,8 +207,12 @@ func (emitter *eventEmitter) EmitEvent(event cadence.Event) error {
 
 		// skip limit if payer is service account
 		// TODO skip only limit-related errors
-		if !isServiceAccount && eventEmitError != nil {
-			return eventEmitError
+		if eventEmitError != nil {
```

added some observability, previously error was swallowed.